### ISSUE TITLE
Assigns inside form buttons without specified type are not updated 

### ISSUE THREAD

**Issue by bartblast** [phoenixframework/phoenix_live_view#1071]
### Environment

* Elixir version (elixir -v): 1.10.3
* Phoenix version (mix deps): 1.5.3
* Phoenix LiveView version (mix deps): 411a5420250be98f19746ceef593c410f805803a
* NodeJS version (node -v): 14.4.0
* NPM version (npm -v): 6.14.5
* Operating system: macOS Catalina
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome 84.0.4147.89 
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: yes

### Actual behavior

Suppose you've got a live component:

```
<form phx-submit="some_event">
  <button><%= @some_assign %></button>
  <label><%= @some_assign %></label>
</form>
```

If the assigns are updated in some handle_event or handle_info callback, the component doesn't use the new value of @some_assign inside the button if the button doesn't specify its type. Tthe label is updated correctly. When the type="button" is specified the problem is fixed.

I understand that this may be the expected behaviour, but if so, it should be documented (I lost a whole day for debugging this).

### Expected behavior

A button without type specified is able to use the new value of @some_assign, or this behaviour is documented.

**Comment by josevalim** [comment]
What is the particular event you are trying? Is it the form submission itself?

**Comment by josevalim** [comment]
So I could not reproduce this. Here is my view:

```elixir
defmodule LiveDemoWeb.PageLive do
  use Phoenix.LiveView

  defmodule Component do
    use Phoenix.LiveComponent

    def render(assigns) do
      ~L"""
      <div>
      <form phx-submit="some_event">
        <button><%= @some_data %></button>
        <label><%= @some_data %></label>
      </form>
      </div>
      <button phx-click="some_event">Other</button>
      """
    end

    def handle_event("some_event", _, socket) do
      {:noreply, assign(socket, :some_data, :rand.uniform(100))}
    end
  end

  def render(assigns) do
    ~L"""
    <%= live_component @socket, Component, some_data: @some_data, id: "stateful" %>
    """
  end

  def mount(_params, _session, socket) do
    {:ok, assign(socket, :some_data, :rand.uniform(100))}
  end

  def handle_event("some_event", _, socket) do
    {:noreply, assign(socket, :some_data, :rand.uniform(100))}
  end
end
```

The form with and without phx-target works just fine. So if you can provide more information, it will be great!

**Comment by bartblast** [comment]
Sure, I created a public repo with the code that I used for debugging:
https://github.com/bartblast/phoenix-liveview-issue-1071

After starting the server,
Visit localhost:4000/basic
and click the top button.
The text inside the button tags won't update.

When you change the button in 
https://github.com/bartblast/phoenix-liveview-issue-1071/blob/master/lib/liveview_problems_web/templates/custom/stateful_component.html.leex

from 
<button phx-click="click" phx-target="<%= @myself %>">BUTTON <%= @text %></button>

to
<button phx-click="click" phx-target="<%= @myself %>" type="button">BUTTON <%= @text %></button>

it will work as "expected".

**Comment by josevalim** [comment]
Thanks, this will be fixed by #1070 as well! See watch it there. :)

**Comment by bartblast** [comment]
Got it, thanks.
